webapp/components/abstract-test-file-results-table.html

```diff
@@ -8,19 +8,30 @@
   <script>
     // eslint-disable-next-line no-unused-vars
     const AbstractTestFileResultsTable = superClass => class extends superClass {
-      static get is() {
-        return 'abstract-test-file-results-table';
-      }
-
       static get properties() {
         return {
           subtestNames: {
             type: Array,
             value: [],
           },
+          isTestHarness: {
+            type: Boolean,
+            computed: 'computeIsTestHarness(subtestNames)',
+          },
         };
       }
 
+      computeIsTestHarness(subtestNames) {
+        // subtestNames is always updated after testRuns[].subtests are updated,
+        // so as long as 'STATUS' exists in subtestNames, we can safely access
+        // testRuns[].subtests['STATUS'].
+        if (subtestNames.includes('STATUS')) {
+          return false;
+        }
+        const statuses = this.testRuns.map(tr => tr.subtests['STATUS'].status);
+        return statuses.includes('OK') || statuses.includes('ERROR');
```

Comment:
This all seems roundabout to me. Why not just check if there are any subtests? And if there aren't, it might be a single harness error, but I think we'd need the manifest to know for sure what kind of test it was? "TIMEOUT" might be either kind, for example.

Reply:
There's always at least a subtest as we currently spoof `testRuns[].subtests` to always include a fake `STATUS` subtest for the file. We could check if there's other non-'STATUS' subtest, but that would fail if the test is a testharness test and all browsers produce harness errors. Eventually we should rely on manifest. The search API may give us the ability to do more useful things on the server side.

Reply:
Where do these data come from? From our sharded individual test files? If so, I believe they contain separate `"status": ...` and `"subtest": [ ... status ... ]`. Perhaps the better long term solution (this may have been in your previous refactor, I'm not sure) would be to disentangle the test status and subtest statuses in whatever component processes data from GCS.

Reply:
... to be clear, I would be fine with filing an issue to handle that and landing this code.